aspnetcore/blazor/javascript-interoperability/import-export-interop.md

```diff
@@ -0,0 +1,378 @@
+---
+title: .NET JavaScript `[JSImport]`/`[JSExport]` interop with ASP.NET Core Blazor WebAssembly
+author: guardrex
+description: Learn how to interact with JavaScript in Blazor WebAssembly apps using .NET JavaScript `[JSImport]`/`[JSExport]` interop.
+monikerRange: '= aspnetcore-7.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 09/28/2022
+uid: blazor/js-interop/import-export-interop
+---
+# .NET JavaScript `[JSImport]`/`[JSExport]` interop with ASP.NET Core Blazor
+
+This article explains how to interact with JavaScript (JS) in Blazor WebAssembly apps using .NET JS `[JSImport]`/`[JSExport]` interop API released with .NET 7.
+
+Blazor provides its own JS interop mechanism based on the <xref:Microsoft.JSInterop.IJSRuntime> interface, which is uniformly supported across Blazor hosting models and described in the following articles:
+
+* <xref:blazor/js-interop/call-javascript-from-dotnet>
+* <xref:blazor/js-interop/call-dotnet-from-javascript>
+
+<xref:Microsoft.JSInterop.IJSRuntime> enables library authors to build JS interop libraries that can be shared across the Blazor ecosystem and remains the recommend approach for JS interop in Blazor.
+
+This article describes an alternative JS interop approach available for the first time with the release of .NET 7. The approaches described in this article can be used in place of existing <xref:Microsoft.JSInterop.IJSRuntime> approaches and should be used to replace obsolete unmarshalled JS interop API when migrating to .NET 7.
```

Comment:
Somewhere we need to provide guidance on when to use the JSImport/JSExport model vs IJSRuntime for both app and library authors. How about this?:

```suggestion
This article describes an alternative JS interop approach specific to WebAssembly-based apps available for the first time with the release of .NET 7. These approaches are appropriate when you only expect to run on WebAssembly and not in the other Blazor hosting models. Library authors can use these approaches to optimize JS interop on WebAssembly by checking at runtime if the app is running on WebAssembly (OperatingSystem.IsBrowser). The approaches described in this article should be used to replace obsolete unmarshalled JS interop API when migrating to .NET 7.
```